### PR TITLE
tests: add savepoint integration tests

### LINKED ISSUE #7387: Add savepoint test
### Which jobs are flaking?

N/A

### Which test(s) are flaking?

N/A

### Jenkins logs or GitHub Actions link

```log
(paste logs here)
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #7387
### What is changed and how it works?

Related PR: https://github.com/pingcap/tidb/pull/34466

#### Tests 

 
 - Integration test
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integration_tests/savepoint/data/prepare.sql [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+--  Check rollback to savepoint will rollback delete.
+delete from t;
+insert into t values (1, 1), (2, 2);
+begin pessimistic;
+insert into t values (3, 3);
+savepoint s1;
+delete from t where id in (1, 3);
+rollback to s1;
+commit;
```
Comment (overvenus) [inline tests/integration_tests/savepoint/data/prepare.sql:158]:
```suggestion
commit;

create table finish_mark (id int PRIMARY KEY);

```

**Review by overvenus — APPROVED** [review]
LGTM, please link to an issue.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**